include/networkit/graph/GraphTools.hpp

```diff
@@ -418,6 +419,63 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Randomizes the weights of the given graph. The weights are uniformly distributed in
+ * the range [0, 1] by default, unless a different distribution is provided. However it
+ * is only strictly in-place for already weighted graphs. For unweighted graphs a copy is
+ * created before randomizing weights.
+ *
+ * @param   G           Directed input graph
+ * @param   distr       Random distribution
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(Graph &G, Distribution distr = std::uniform_real_distribution<edgeweight>{
+                                    0, std::nexttoward(1.0, 2.0)}) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+#pragma omp parallel
+    {
+        std::mt19937 gen;
+        auto baseSeed = Aux::Random::integer();
+
+#pragma omp for schedule(static)
+        for (omp_index u = 0; u < G.upperNodeIdBound(); u++) {
+            // we use the hash of the origin node so that the weights are independent
+            // of  processing order
+            gen.seed(std::hash<node>()(u) + 0x9e3779b9 + (baseSeed << 6) + (baseSeed >> 2));
+            index j = 0;
+            index k;
+            edgeweight ew;
```

Comment:
You can move these two inside the loop, there is no advantage in declaring them here.